src/bika/lims/browser/analyses/view.py

```diff
@@ -1122,6 +1190,19 @@ def _folder_item_instrument(self, analysis_brain, item):
         else:
             item["Instrument"] = _("Manual")
 
+
+    def _on_unit_change(self, uid=None, value=None, item=None, **kw):
+        """ updates the rendered unit on selection of unit. 
+        """
```

Comment:
Note that value and item are declared as `None` in the function signature. Thus, better to add a check like the following before anything else:

```python
        if not all([value, item]):
            return None
```

Reply:
Thank you, this is now included.